### PR TITLE
Remove multiple paths from ShardPath

### PR DESCRIPTION
**Pull request by rjernst**
relates #71205

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/shard/ShardPath.java [hunk 1]
```diff
@@ -107,51 +107,38 @@ public static ShardPath loadShardPath(Logger logger, NodeEnvironment env,
...
         final Path sharedDataPath = env.sharedDataPath();
-        return loadShardPath(logger, shardId, customDataPath, new Path[] { shardPath }, sharedDataPath);
+        return loadShardPath(logger, shardId, customDataPath, shardPath, sharedDataPath);
     }
 
     /**
      * This method walks through the nodes shard paths to find the data and state path for the given shard. If multiple
      * directories with a valid shard state exist the one with the highest version will be used.
      * <b>Note:</b> this method resolves custom data locations for the shard.
      */
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/index/shard/ShardPath.java:117]: This comment looks out of date now.

File: server/src/main/java/org/elasticsearch/index/shard/ShardPath.java [hunk 2]
```diff
@@ -107,51 +107,38 @@ public static ShardPath loadShardPath(Logger logger, NodeEnvironment env,
...
-    public static ShardPath loadShardPath(Logger logger, ShardId shardId, String customDataPath, Path[] availableShardPaths,
+    public static ShardPath loadShardPath(Logger logger, ShardId shardId, String customDataPath, Path availableShardPath,
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/index/shard/ShardPath.java:118]: Isn't this just `shardPath` now?

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left a few comments.

File: server/src/main/java/org/elasticsearch/index/shard/ShardPath.java (hunk 1 above)
Comment (rjernst) [reply server/src/main/java/org/elasticsearch/index/shard/ShardPath.java:115]: Good catch. I tweaked these in 16975ec

File: server/src/main/java/org/elasticsearch/index/shard/ShardPath.java (hunk 2 above)
Comment (rjernst) [reply server/src/main/java/org/elasticsearch/index/shard/ShardPath.java]: Yes. Renamed in 16975ec

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**